misc-utils/rename.c

```diff
@@ -77,18 +77,21 @@ static int do_symlink(char *from, char *to, char *s, int verbose, int noact, int
 	if (string_replace(from, to, target, target, &newname))
 		ret = 0;
 
-	if (ret == 1 && nooverwrite && lstat(target, &sb) == 0) {
-		if (verbose)
-			printf(_("Skipping existing link: `%s' -> `%s'\n"), s, target);
-
-		ret = 0;
-	}
-
 	if (ret == 1) {
-		if (!noact && 0 > unlink(s)) {
+		if (access(s, F_OK) != 0) {
+			warn(_("%s: not accessible"), s);
+			ret = 2;
+		}
+		else if (nooverwrite && lstat(target, &sb) == 0) {
+			if (verbose)
+				printf(_("Skipping existing link: `%s'\n"), target);
```

Comment:
It seems you have reverted "%s -> %s" here. I guess it's unwanted change. Right?